### ISSUE TITLE
Setup Error

### ISSUE THREAD

**Issue by hiteshgarg14** [opencivicdata/python-opencivicdata-django#80]
While setting up `python-opencivicdata-django` for `openstates/openstates-pupa` I got this error.

```
Obtaining python-opencivicdata from git+https://github.com/opencivicdata/python-opencivicdata-django.git#egg=python-opencivicdata
  Cloning https://github.com/opencivicdata/python-opencivicdata-django.git to /opt/openstates/venv-pupa/src/python-opencivicdata
```
```diff
- Running setup.py (path:/opt/openstates/venv-pupa/src/python-opencivicdata/setup.py) egg_info for package python-opencivicdata produced metadata for project name opencivicdata-django. Fix your #egg=python-opencivicdata fragments.
```
```
Collecting Django>=1.9 (from opencivicdata-django)
  Downloading Django-1.10.6-py2.py3-none-any.whl (6.8MB)
Collecting psycopg2 (from opencivicdata-django)
  Downloading psycopg2-2.7.1-cp35-cp35m-manylinux1_x86_64.whl (2.7MB)
Collecting opencivicdata-divisions (from opencivicdata-django)
  Downloading opencivicdata-divisions-2017.3.22.tar.gz (2.4MB)
Building wheels for collected packages: opencivicdata-divisions
  Running setup.py bdist_wheel for opencivicdata-divisions: started
  Running setup.py bdist_wheel for opencivicdata-divisions: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/d4/16/22/543c321bb47ba44dc64843b65e519d407c1a7b9ba50cbd2175
Successfully built opencivicdata-divisions
Installing collected packages: Django, psycopg2, opencivicdata-divisions, opencivicdata-django
  Running setup.py develop for opencivicdata-django
    Complete output from command /opt/openstates/venv-pupa/bin/python3 -c "import setuptools, tokenize;__file__='/opt/openstates/venv-pupa/src/python-opencivicdata/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/setuptools/__init__.py", line 12, in <module>
        import setuptools.version
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/setuptools/version.py", line 1, in <module>
        import pkg_resources
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2927, in <module>
        @_call_aside
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2913, in _call_aside
        f(*args, **kwargs)
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2952, in _initialize_master_working_set
        add_activation_listener(lambda dist: dist.activate())
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/pkg_resources/__init__.py", line 956, in subscribe
        callback(dist)
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2952, in <lambda>
        add_activation_listener(lambda dist: dist.activate())
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2515, in activate
        declare_namespace(pkg)
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2097, in declare_namespace
        _handle_ns(packageName, path_item)
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2047, in _handle_ns
        _rebuild_mod_path(path, packageName, module)
      File "/opt/openstates/venv-pupa/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2066, in _rebuild_mod_path
        orig_path.sort(key=position_in_sys_path)
    AttributeError: '_NamespacePath' object has no attribute 'sort'
    
    ----------------------------------------

 Command "/opt/openstates/venv-pupa/bin/python3 -c "import setuptools, tokenize;__file__='/opt/openstates/venv-pupa/src/python-opencivicdata/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps" failed with error code 1 in /opt/openstates/venv-pupa/src/python-opencivicdata/
The command '/bin/sh -c /opt/openstates/venv-pupa/bin/pip install -e git+https://github.com/opencivicdata/python-opencivicdata-django.git#egg=python-opencivicdata' returned a non-zero code: 1

```

**Comment by jamesturk** [comment]
Hmm.. this is probably related to the Py2.7 work, can you verify that this
doesn't occur if you try checking out @ca8217f?

**Comment by jamesturk** [comment]
Also, @gordonje if you have a chance maybe you can help address?

**Comment by hiteshgarg14** [comment]
@jamesturk I tried it by checking out  @ ca8217f and it worked perfectly :+1:

**Comment by gordonje** [comment]
Looking into this now.

@hiteshgarg14 Can you tell me which specific version of Python 3.5 and setuptools you have?

**Comment by hiteshgarg14** [comment]
@gordonje 
 - Python 3.5.2 
 - setuptools 34.3.2

**Comment by jaraco** [comment]
See pypa/pip#4216

**Comment by gordonje** [comment]
Looks like the problem is in pip, as indicated by the issue referenced by @jaraco (thanks for your help in trying to resolve this).

Weirdly enough, though, I have not been able to reproduce the @hiteshgarg14's error. What I've tried so far is setting up a virtualenv and then `pip install -e git+https://github.com/opencivicdata/python-opencivicdata-django.git#egg=python-opencivicdata`. Did not get this error with that version of Python and setuptools. Also could didn't get it with setuptools 34.3.3.

Can you also tell me which version of pip you have? I'm currently running 9.0.1

And just so I'm clear: Your running the setup process via Docker as outlined in the [openstates repo](https://github.com/openstates/openstates)?

**Comment by fgregg** [comment]
from that linked pip issue if you upgrade setuptools, then the problem is resolved?

**Comment by hiteshgarg14** [comment]
@gordonje the pip version is `9.0.1` and yeah! I got the error while rebuilding the docker image with more dependencies. I verified python, setuptools and pip versions via docker bash after activating `venv-pupa`.

**Comment by jaraco** [comment]
The problem exists in Setuptools, which was fixed (or at least patched) in 31.0.1, but pip still bundles an old version which is still broken. There's no known version of pip that's not broken. You might try installing pip from master.

**Comment by jamesturk** [comment]
We may need to roll this back for now, until there's a pip release at least

**Comment by jamesturk** [comment]
We install from source in a lot of places so I'd rather have master fixed & we can reintroduce 2.7 support on a branch for now.

This is affecting our docker builds, so I'm going to work on a fix today.

https://hub.docker.com/r/openstates/openstates-pupa/builds/bvebk3wotxxaote8giyoy8x/

**Comment by jamesturk** [comment]
OK, this is backed out, we can add this namespace package in once again once we've verified it is working on Python 3 too.  Sorry for the trouble, everything else Py2 related is intact here so should be easy to re-add once we're ready
